### PR TITLE
test(signer): Generate cycles depositor pic bindings

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
Signer tests need to make calls to various canisters installed on pocket_ic.  The cycles depositor is the first of these canisters.

# Changes
- Add a didc command to generate `cycles_depositor` bindings.
  - Note: a custom template is needed; the default bindings cannot be applied to pocketic.
- Added a CI job to verify that bindings are up to date
- Updated bindings, because they were not up to date.

# Tests
- A CI check is included that the bindings are up to date.

### CODE REVIEW

File: scripts/bind/pic/cycles_depositor.hbs
Comment (bitdivine) [inline scripts/bind/pic/cycles_depositor.hbs:1]: Note: This is the handlebars template used by didc to generate the Rust pocket-ic bindings.

Note: I assume that this template will work for other canisters as well but I want to be sure before I rename this to something not canister-specific.

File: .github/actions/bindings/action.yaml
Comment (bitdivine) [inline .github/actions/bindings/action.yaml:1]: An action that checks that the bindings are up to date.

File: .github/workflows/check.yml
Comment (bitdivine) [inline .github/workflows/check.yml:1]: Here we invoke the action that checks the bindings.  We free-ride on the fact that dfx deploy gets all the candid files, so we don't need to.

File: dev-tools.json
Comment (bitdivine) [inline dev-tools.json:1]: Specification for where to get `didc`.  Note that the download URL and the version are unrelated.  Earlier download URLs with the same version don't support templated `didc bind`.

File: src/signer/canister/tests/it/canister/cycles_depositor.rs
Comment (bitdivine) [inline src/signer/canister/tests/it/canister/cycles_depositor.rs:1]: This is the main output of this PR:  Canister bindings for use with pocket-ic!  I started with the simplest canister so it has just an unimpressive single method.  The others are more complicated.

File: src/signer/canister/signer.did
Comment (bitdivine) [inline src/signer/canister/signer.did:1]: Looks like the candid file was out of date :facepalm:   Now we have a check to catch this.

File: src/signer/canister/tests/it/utils/cycles_depositor.rs
Comment (lmuntaner) [inline src/signer/canister/tests/it/utils/cycles_depositor.rs:1]: Is this a copy of the one in `tests/it/canister/`?

**Review by lmuntaner — APPROVED** [review]
Very cool PR!

File: scripts/bind/pic/cycles_depositor.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+didc bind -t rs "$(jq -r .canisters.cycles_depositor.candid dfx.json)" --config scripts/bind/pic/cycles_depositor.toml >src/signer/canister/tests/it/canister/cycles_depositor.rs
```
[the remark this answers is not in the document]
Comment (bitdivine) [reply scripts/bind/pic/cycles_depositor.sh:2]: Asking the languages team & reading the docs.  The docs are here: https://github.com/dfinity/candid/blob/master/spec/Type-selector.md#rust-binding-configuration and https://github.com/dfinity/candid/tree/master/tools/didc#toml-config.

[the remark this answers is not in the document]
Comment (bitdivine) [reply scripts/bind/pic/cycles_depositor.sh:2]: Re SNS; I'm not sure I understand exactly what you mean but we can sync in person and probably find a solution.

File: scripts/bind/pic/cycles_depositor.toml
[the remark this answers is not in the document]
Comment (bitdivine) [reply scripts/bind/pic/cycles_depositor.toml:1]: The docs are here: https://github.com/dfinity/candid/blob/master/spec/Type-selector.md#rust-binding-configuration and https://github.com/dfinity/candid/tree/master/tools/didc#toml-config.

For new features like this it's worth just watching the codebase.

File: src/signer/canister/tests/it/utils/cycles_depositor.rs
Comment (bitdivine) [reply src/signer/canister/tests/it/utils/cycles_depositor.rs:1]: Good catch.  I moved it into the canister directory but didn't delete the old one.

File: src/signer/canister/tests/it/canister/cycles_depositor.rs
Comment (bitdivine) [reply src/signer/canister/tests/it/canister/cycles_depositor.rs:1]: Correct.  I use it in a branch but not here.

**Pull request merged**